### ISSUE TITLE
Azure Function fails to load System.Configuration.ConfigurationManager

### ISSUE THREAD

**Issue by fgheysels** [dotnet/runtime#54642]
### Description

I have an Azure Function which imports the NuGet Package `System.Configuration.ConfigurationManager` v5.0.0
The Azure Function is a .NET Core 3.1 app and targets Azure Functions v3.

When I start the function, it crashes with a FileNotFoundException:

```
 System.IO.FileNotFoundException: Could not load file or assembly 'System.Configuration.ConfigurationManager, Version=5.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. The system cannot find the file specified.
File name: 'System.Configuration.ConfigurationManager, Version=5.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'
```
When I look at my filesystem, I notice that the assembly System.Configuration.ConfigurationManager.dll is not present in the `netcoreapp3.1\bin` folder, whereas all my other assemblies that I require in my function are there.
However, the assembly is present in the `netcoreapp3.1` folder.

When I downgrade to a previous version (4.7.0), I don't have this issue. (I can also see that the assembly System.Configuration.ConfigurationManager.dll is now present in the `netcoreapp3.1\bin` folder.

### Configuration

.NET Core 3.1
Azure Functions v3
Targetting a Linux App Service

### Regression?

As mentionned in the description, it works with version 4.7.0 of System.Configuration.ConfigurationManager in a .NET Core 3.1 app.

### Other information

Probably related to https://github.com/dotnet/runtime/issues/627

**Comment by krwq** [comment]
@fgheysels can you try with 6.0-preview?

**Comment by fgheysels** [comment]
I just tried it with 6.0.0-preview.5.21301.5, and ran into the same error:

![image](https://user-images.githubusercontent.com/3605786/125444040-032d886f-3166-44da-899f-a8905bc72a48.png)

**Comment by shafaqat-ali-cms365** [comment]
I am also getting this same error. V 5.0.0

**Comment by joperezr** [comment]
How are you publishing your application? I'm not able to reproduce the issue. The ConfigurationManager package has a netstandard2.0 ref and lib files which should be used when targeting netcoreapp3.1, so I think that the error here is more related to your deployment or setup than to the actual package itself. Can you share more info on how do you deploy, and how do you build your application?

**Comment by shafaqat309** [comment]
> How are you publishing your application? I'm not able to reproduce the issue. The ConfigurationManager package has a netstandard2.0 ref and lib files which should be used when targeting netcoreapp3.1, so I think that the error here is more related to your deployment or setup than to the actual package itself. Can you share more info on how do you deploy, and how do you build your application?

I am using .netcore3.1 and publishing through VS to Azure function apps

**Comment by joperezr** [comment]
Can you share a minimal repro project so that I can take a peek? This may just be a problem with VS publish to Azure.

**Comment by Trinitron** [comment]
> Can you share a minimal repro project so that I can take a peek? This may just be a problem with VS publish to Azure.

Paste these lines to any v2 function (netcoreapp3.1). There is no need for publishing, it fails locally. 
`            var codeBase = System.Reflection.Assembly.GetExecutingAssembly().CodeBase;
            var binDir = System.IO.Path.GetDirectoryName(codeBase).Replace(@"file:\", "");
            System.Configuration.ExeConfigurationFileMap configFileMap = new System.Configuration.ExeConfigurationFileMap() { ExeConfigFilename = binDir + @"\App.config" };
            Configuration config2 = System.Configuration.ConfigurationManager.OpenMappedExeConfiguration(configFileMap, ConfigurationUserLevel.None);`

**Comment by fgheysels** [comment]
> Can you share a minimal repro project so that I can take a peek? This may just be a problem with VS publish to Azure.

You can already reproduce it running locally.
Just create an Azure Function, reference the package and run locally.  

I've created a sample repo [here](https://github.com/fgheysels/FunctionApp_ConfigurationManagerIssue).  This sample consists of a timer-triggered Azure Function that throws a `ConfigurationErrorsException` on each invocation.   You'll see that, instead of the ConfigurationErrorsException being thrown, a `FileNotFoundException` is thrown.

**Comment by ajkimberley** [comment]
I'm also experiencing this issue, on both .NET 5.0 and 6.0-preview. I found downgrading the System.Confguration.ConfigurationManager from 5.0.0 to 4.7.0 resolves the issue - though of course that's not ideal.

**Comment by fijsh** [comment]
@joperezr i think this should be reopened since this also occurs on .NET6 (see above)
I have the same issue on .NET Core 3.1 when I try to use Microsoft.Data.SqlClient. With System.Data.SqlClient there is no issue.

**Comment by fijsh** [comment]
> @joperezr i think this should be reopened since this also occurs on .NET6 (see above) I have the same issue on .NET Core 3.1 when I try to use Microsoft.Data.SqlClient. With System.Data.SqlClient there is no issue.

Just got a breakthrough for my issue. The System.Configuration.ConfigurationManager.dll was not in the netcoreapp3.1/bin folder, but it was in the netcoreapp3.1 folder. After I manually copied the System.Configuration.ConfigurationManager.dll in the bin folder I got the same error on the file System.Text.Encoding.CodePages.dll. I also copied that file to the bin folder and then it worked.

What is exactly causing this? I never had this before. I'm not sure yet what happens if I deploy to Azure.

**Comment by fgheysels** [comment]
@fijsh this is also what I've observed and I mentionned that in the issue.
However, imho it is not sustainable to work around this by hacking some copy statements in your CI/CD pipeline if you have automated deployments (imho).

**Comment by fijsh** [comment]
> @fijsh this is also what I've observed and I mentionned that in the issue.
> 
> However, imho it is not sustainable to work around this by hacking some copy statements in your CI/CD pipeline if you have automated deployments (imho).

You are right. I was linked to this topic and somehow didn't read your post. Indeed not a solution to hack this in CI/CD. I'm considering moving back to system.data.sqlclient. Too bad that I can't use the retry provider then. This bug should be fixed asap.

**Comment by Aileron79** [comment]
Same problem here, I have a core 3.1 function app which used to work fine. The project included the Z.EntityFramework.Extensions.EFCore package which I uninstalled and replaced with EFCore.BulkExtensions. After changing the bulk insert operations in the code accordingly, when I compiled and ran the project locally, it complained about the configuration manager missing. Copying the file did not solve the issue, Downgrading the ConfigurationManager from 6.0.0 to 4.7.0 solved the issue - which clearly is not the way it is supposed to be solved as it used to work with 6.0.0 before.

**Comment by joperezr** [comment]
We could reopen this issue if desired, but I don't think this is the way to go. The reason is that the specific problem that I believe you are hitting is NOT related to the NuGet package itself, so reopening the issue on this repo (which is only in charge of building said package) wouldn't really do much as the potential fix for this issue wouldn't happen here. My suggestion instead, is to log an issue directly to the Azure functions team which are the ones that can diagnose/fix/resolve the issue.

**Comment by fgheysels** [comment]
I have created a new issue in the Azure-Functions repo [here](https://github.com/Azure/Azure-Functions/issues/2116)
